### ISSUE TITLE
Session saving problem

### ISSUE THREAD

**Issue by AlexNasonov** [mweibel/connect-session-sequelize#38]
I get an error each time I try to save Session to db
```
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:346:11)
```
Which is perfectly seen if you try to save even uninitialized session like me:
```
app.use(session({
    genid: function () {
        return uuid.v1();
    },
    secret: 'marvin-erp',
    resave: false,
    saveUninitialized: true,
    checkExpirationInterval: 15 * 60 * 1000, // The interval at which to cleanup expired sessions in milliseconds.
    expiration: 24 * 60 * 60 * 1000,  // The maximum age (in milliseconds) of a valid session.
    store: new SequelizeStore({
        db: sequelize
    })
}));
```
This problem started since 3.1.0 version.
I'm using Postgres if it helps)

**Comment by mweibel** [comment]
thanks for the bug report. Are you sure only the `connect-session-sequelize` version changed since that error appears? 
If you try downgrading to 3.0.0 the error goes away?

**Comment by AlexNasonov** [comment]
I have tried different options. My app saves and retrieves any data from db normally, I even tried different versions of sequelize (up to recent 4.0). I also created a new node from scratch with only few neccessary dependencies:
```
"dependencies": {
    "body-parser": "~1.13.2",
    "connect-session-sequelize": "^3.1.0",
    "cookie-parser": "~1.3.5",
    "debug": "~2.2.0",
    "ejs": "~2.3.3",
    "express": "~4.13.1",
    "express-session": "^1.14.0",
    "morgan": "~1.6.1",
    "nconf": "^0.8.4",
    "node-uuid": "^1.4.7",
    "pg": "^6.0.1",
    "pg-hstore": "^2.3.2",
    "sequelize": "git+https://git@github.com/sequelize/sequelize.git",
    "serve-favicon": "~2.3.0"
  }
```
And still get this error. Hope you'll help me find out what's the problem, because it denies any user login possibilities((

**Comment by mweibel** [comment]
if you try without the store, i.e. use the default memory store it works?

**Comment by AlexNasonov** [comment]
If I do it without store in's OK

**Comment by AlexNasonov** [comment]
connect-session-sequelize.js
str.92:
`return session.save().return(data);`

I tried to change it to:
```
 return this.sessionModel.findOrCreate({where: {'sid': sid}, defaults: defaults}).spread(function sessionCreated(session) {
      if(session['data'] !== stringData) {
        session['data'] = JSON.stringify(data);
        session['expires'] = expires;
        return session.save().then(function (data) {
          return(data);
        });
      }
      else return session;
    }).asCallback(fn);
```
And it worked.

Also, I recommend to change session model sid limit, from 32 to 64 (for example) as UUID has more then 32 limits.

**Comment by jwoos** [comment]
I am also seeing this problem! I'll try the suggested fix and report back

**Comment by mweibel** [comment]
Please check if it's working better now. The fix for #41 also changed something in that regard.
